lib/test_runner.zig

```diff
@@ -81,6 +81,13 @@ pub fn main() void {
             ok_count += 1;
             test_node.end();
             if (!have_tty) std.debug.print("OK\n", .{});
+            // Workaround issue #1923 by clearing error stack traces between tests
+            //
+            // This is not a true fix, but helps isolate errors to the tests where they
+            // originate instead of confusing them all together
+            if (@errorReturnTrace()) |trace| {
+                trace.index = 0;
+            }
```

Comment:
Could this possibly be moved outside of the `if`? Haven't fully thought it through, but it seems like resetting the index even after failed tests could make sense as well. I'll try to come up with a case where it matters and report back.

Reply:
I think it should be moved outside of the if (below the else condition). Here's a simple case that shows the problem:

```zig
test "test skipped" {
    return error.SkipZigTest;
}

test "trigger error trace" {
    return error.WeExpectToSeeThisInTheTrace;
}
```

With the index resetting inside the `if` block:

```
Test [1/2] test "test expected error"... SKIP
Test [2/2] test "trigger error trace"... FAIL (WeExpectToSeeThisInTheTrace)
/home/ryan/Programming/zig/tmp/1923.zig:8:5: 0x208152 in test "test expected error" (test)
    return error.SkipZigTest;
    ^
/home/ryan/Programming/zig/tmp/1923.zig:12:5: 0x208172 in test "trigger error trace" (test)
    return error.WeExpectToSeeThisInTheTrace;
    ^
0 passed; 1 skipped; 1 failed.
```

and with it after the `else` block:

```
Test [1/2] test "test skipped"... SKIP
Test [2/2] test "trigger error trace"... FAIL (WeExpectToSeeThisInTheTrace)
/home/ryan/Programming/zig/tmp/1923.zig:6:5: 0x208172 in test "trigger error trace" (test)
    return error.WeExpectToSeeThisInTheTrace;
    ^
0 passed; 1 skipped; 1 failed.
```